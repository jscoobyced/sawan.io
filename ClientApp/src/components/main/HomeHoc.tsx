import React from 'react';
import { ContentServiceFactory } from '../../services/ContentServiceFactory';
import { IContentService } from '../../services/IContentService';
import { BlogPage } from '../../services/Models';
import { Home } from './Home';

export interface HomeState {
    blogPage: BlogPage;
}

export class HomeHoc extends React.Component<{}, HomeState> {
    private readonly contentService: IContentService;
    private readonly numberOfElements = 3;

    public constructor(props: {}) {
        super(props);
        this.contentService = ContentServiceFactory.GetContentService();
        this.state = {
            blogPage: this.contentService.getDefaultBlogPage()
        };
    }

    public componentDidMount() {
        this.contentService.getBlogPage(this.numberOfElements)
            .then(blogPage => {
                const articles = blogPage.articles.slice(0, this.numberOfElements);
                this.setState({
                    blogPage: {
                        articles
                    }
                });
            });
    }

    public render() {
        return <Home blogPage={this.state.blogPage} />;
    }
}
