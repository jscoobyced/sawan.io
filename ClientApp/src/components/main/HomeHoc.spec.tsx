import { shallow } from 'enzyme';
import React from 'react';
import { HomeHoc } from './HomeHoc';

test('HomeHoc component is unchanged.', async () => {
    const homeHoc = shallow(<HomeHoc />);
    expect(homeHoc).toMatchSnapshot();
});
