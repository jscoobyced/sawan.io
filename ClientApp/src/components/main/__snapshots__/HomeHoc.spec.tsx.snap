// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHoc component is unchanged. 1`] = `
<Home
  blogPage={
    Object {
      "articles": Array [],
    }
  }
/>
`;
